File: Low Level Design (LLD)/DesignPatterns/1. Creational/Prototype/Java/TShirt.java
public class TShirt implements Product {
    private String size;
    private String color;

    public TShirt(String size, String color) {
        this.size = size;
        this.color = color;
    }

    @Override
    public Product clone() {
        return new TShirt(this.size, this.color);
    }

    @Override
    public void showDetails() {
        System.out.println("T-Shirt [Size: " + size + ", Color: " + color + "]");
    }

    // Optional: Setters for customization
    public void setSize(String size) {
        this.size = size;
    }

    public void setColor(String color) {
        this.color = color;
    }
}

File: Low Level Design (LLD)/DesignPatterns/1. Creational/Prototype/Cpp/TShirt.cpp
#include <iostream>
#include "TShirt.h"

TShirt::TShirt(const std::string& size, const std::string& color)
    : size(size), color(color) {}

Product* TShirt::clone() const {
    return new TShirt(size, color); // shallow clone (primitive types)
}

void TShirt::showDetails() const {
    std::cout << "T-Shirt [Size: " << size << ", Color: " << color << "]\n";
}

void TShirt::setSize(const std::string& newSize) {
    size = newSize;
}

void TShirt::setColor(const std::string& newColor) {
    color = newColor;
}
